File: app/backend/NewerDown.Domain/Entities/Monitor.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using NewerDown.Domain.Enums;

namespace NewerDown.Domain.Entities;

public class Monitor
{
    public Guid Id { get; set; }
    
    public string Name { get; set; }
    
    public string Target { get; set; } = default!;
    
    public int? Port { get; set; }
    
    public MonitorType Type { get; set; }
    
    public int IntervalSeconds { get; set; }
    
    public bool IsActive { get; set; }
    
    public DateTime CreatedAt { get; set; }
    
    public Guid? UserId { get; set; }
    
    public User User { get; set; }
    
    public ICollection<MonitorCheck> Checks { get; set; } = new List<MonitorCheck>();
    
    public ICollection<Alert> Alerts { get; set; } = new List<Alert>();
}

public sealed class MonitorConfiguration : IEntityTypeConfiguration<Monitor>
{
    public void Configure(EntityTypeBuilder<Monitor> builder)
    {
        builder.HasOne(m => m.User)
            .WithMany(u => u.Monitors)
            .HasForeignKey(m => m.UserId)
            .OnDelete(DeleteBehavior.SetNull);
    }
}

File: app/frontend/src/shared/types/Monitor.ts
export enum MonitorType {
  HTTP = 0,
  HTTPS = 1,
  TCP = 2,
  PING = 3,
}

export interface Monitor {
  id: string;
  name: string;
  url: string;
  checkIntervalSeconds: number;
  isActive: boolean;
  type: MonitorType;
  createdAt: string;
  userId: string;
}

export interface CreateMonitorRequest {
  name: string;
  target: string;
  type: MonitorType;
  port: number;
  intervalSeconds: number;
  isActive: boolean;
}

export interface DeleteMonitorRequest {
  id: string;
}

export interface MonitorsState {
  monitors: Monitor[];
  loading: boolean;
  error: string | null;
  hasMore: boolean;
}
